storybookjs/storybook
# UI: Fix custom theme hover-color inconsistency

Issue: #22199



## What I did
- Currently, a tint of the default color.secondary in base.ts (code/lib/theming/src/base.ts) is used as background color in the bar, so changed to use theme.color.secondary as background color when BranchNode is hovered or focused.
- The same applies to LeafNode



※ The reason why 0.93 is passed for the first argument of the transparentize function is because the value of background.hoverable in base.ts (code/lib/theming/src/base.ts) did so.

| BranchNode(before)| BranchNode(after) |LeafNode(before) |LeafNode(after) |
| :---: | :---: | :---: | :---: |
| ![スクリーンショット 2023-04-26 23 18 57](https://user-images.githubusercontent.com/57059705/234606019-7fc0dc0c-86b5-4866-99d6-eb490865e0cb.png)|<img width="212" alt="スクリーンショット 2023-04-26 23 24 29" src="https://user-images.githubusercontent.com/57059705/234606591-cdb6c945-0909-4615-bfc1-4ceb975693ac.png"> |![スクリーンショット 2023-04-26 23 18 09](https://user-images.githubusercontent.com/57059705/234605636-65514e95-f2ad-4b56-ac3e-9c5a9715a6c9.png) |<img width="213" alt="スクリーンショット 2023-04-26 23 25 42" src="https://user-images.githubusercontent.com/57059705/234607019-36cc4fee-5214-4c45-8540-91c12f965c24.png"> |


## How to test
- generate a new theme using the create() function from storybook/theming as folows.
```ts
// code/ui/.storybook/manager.ts
import { addons } from '@storybook/manager-api';
import startCase from 'lodash/startCase.js';
import { create } from '@storybook/theming/create';

addons.setConfig({
  sidebar: {
    renderLabel: ({ name, type }) => (type === 'story' ? name : startCase(name)),
  },
  theme: create({
    base: 'light',
    colorSecondary: '#FF8688',
  }),
});
```
- `cd code`
- `yarn storybook:ui`
- Check that the secondary-based color is used as background color when the BranchNode/LeafNode is hovered or focused.



## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
